Rename variables, dimens

Valentindp opened the pull request.

Valentindp committed:
1. Добавлены постоянные переменные в dimens
2. Убрал лишние переменные. Kotlin Android Extensions
3. Singleton CityRepository через object
4. Поиск объекта в списке и установка нужного статуса

Valentindp committed:
Изменены названия переменных

severianremi commented on app/src/main/java/valentyn/androidtasks/views/CityActivity.kt:
```diff
@@ -70,11 +69,17 @@ class CityActivity : AppCompatActivity() {
     }
 
     override fun onSupportNavigateUp(): Boolean {
```
For Activity, you can use onBackPressed()

Valentindp committed:
Add tabs and fragments
New repository - ParkRepository

severianremi commented on app/src/main/AndroidManifest.xml:
```diff
@@ -17,7 +17,8 @@
             </intent-filter>
         </activity>
-        <activity android:name=".views.CityActivity"/>
+        <activity android:name=".views.CityActivity"></activity>
```
You should add <orientation> tag if you don't have requirements to handle both orientations =)

severianremi commented on app/src/main/java/valentyn/androidtasks/adapters/FragmentsPagerAdapter.kt:
```diff
@@ -0,0 +1,38 @@
+
+    override fun getPageTitle(position: Int): CharSequence {
+        return when (position) {
+            0 -> "Сities"
```
Please, use context.getStrings()

severianremi replied:
Please, read the different between FragmentStatePagerAdapter and FragmentPagerAdapter. And say what the best solution for your code

Valentindp replied:
The best solution for my task would be FragmentPagerAdapter.
FragmentPagerAdapter needed for a small number of fragments

severianremi commented on app/src/main/java/valentyn/androidtasks/adapters/ParksAdapter.kt:
```diff
@@ -0,0 +1,57 @@
+
+class ParksAdapter(private val mDataset: List<Park>, val clickListener: (Park) -> Unit) :
+    RecyclerView.Adapter<ParksAdapter.ParkViewHolder>() {
+
```
empty space =(

Valentindp replied:
Не понял про "место", класс создан корректно, что не так?

severianremi replied:
Лишняя пустая строчка. Убери ее

severianremi commented on app/src/main/java/valentyn/androidtasks/adapters/ParksAdapter.kt:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    override fun onBindViewHolder(holder: ParkViewHolder, position: Int) {
+
```
reformat code, plese

severianremi commented on app/src/main/java/valentyn/androidtasks/adapters/ParksAdapter.kt:
```diff
@@ -0,0 +1,57 @@
+
+    class ParkViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
+
+        private var Park: Park? = null
```
wrong naming =(

severianremi replied:
Зачем тебе здесь переменная глобальная?

severianremi commented on app/src/main/java/valentyn/androidtasks/adapters/ParksAdapter.kt:
```diff
@@ -0,0 +1,57 @@
+
+        private var Park: Park? = null
+
+        fun bindData(Park: Park, clickListener: (Park) -> Unit) {
```
названия!! Пожалуйста, придерживайся конвенций. ЭТО ВАЖНО!

severianremi commented on app/src/main/java/valentyn/androidtasks/adapters/ParksAdapter.kt:
```diff
@@ -0,0 +1,57 @@
+
+    }
+
+    override fun getItemCount() = mDataset.size
```
В kotlin не используют венгерскую аннотацию

severianremi commented on app/src/main/java/valentyn/androidtasks/fragments/CityFragment.kt:
```diff
@@ -0,0 +1,52 @@
+
+class CityFragment : Fragment()  {
+
+    private val mDataset: List<City> = CityRepository.DataCitys
```
неправильное название

severianremi commented on app/src/main/java/valentyn/androidtasks/fragments/CityFragment.kt:
```diff
@@ -0,0 +1,52 @@
+
+        val view = inflater.inflate(R.layout.fragment_city, container, false)
+
+        view.city_recyclerView.apply {
```
используй kotlin extentions тебе не нужен переменная val view

severianremi commented on app/src/main/java/valentyn/androidtasks/fragments/CityFragment.kt:
```diff
@@ -0,0 +1,52 @@
+            layoutManager = LinearLayoutManager(activity)
+            adapter = CitysAdapter(mDataset, { city: City -> cityClicked(city) })
+        }
+        return view //super.onCreateView(inflater, container, savedInstanceState)
```
почему ты закоментировал //super.onCreateView(inflater, container, savedInstanceState)??

Valentindp replied:
Насколько я понял функция onCreateView во фрагменте должна возвращать view с конкретного layout. Возвращая onCreateView  суперкласса не рисует.
В методе  onCreateView не получиться использовать kotlin extensions так как на тот момент еще нет view (все расширения null)
Отрисовку списка перенес в onViewCreated - там уже работает расширение. Думаю и архитектурно так правильней

severianremi commented on app/src/main/java/valentyn/androidtasks/fragments/ParkFragment.kt:
```diff
@@ -0,0 +1,51 @@
+            layoutManager = LinearLayoutManager(activity)
+            adapter = ParksAdapter(mDataset, { park: Park -> cityClicked(park) })
+        }
+        return view//super.onCreateView(inflater, container, savedInstanceState)
```
Замечания те же, что и в классе выше

severianremi commented on app/src/main/java/valentyn/androidtasks/models/Park.kt:
```diff
@@ -0,0 +1,34 @@
+            var about: String,
+            var country: String,
+            var site: String,
+            var select: Boolean) : Serializable
```
Замени на Parcelable

severianremi commented on app/src/main/java/valentyn/androidtasks/views/ParkActivity.kt:
```diff
@@ -0,0 +1,89 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_park)
```
Слишком длинный метод - разбей его на более маленькие

severianremi commented on app/src/main/java/valentyn/androidtasks/views/ParkActivity.kt:
```diff
@@ -0,0 +1,89 @@
+        setContentView(R.layout.activity_park)
+
+        setSupportActionBar(toolbar_park)
+        getSupportActionBar()?.setDisplayShowTitleEnabled(false)
```
getSupportActionBar()?.apply{}

severianremi commented on app/src/main/java/valentyn/androidtasks/views/ParkActivity.kt:
```diff
@@ -0,0 +1,89 @@
+            true
+        }
+        android.R.id.home ->{
+            onSupportNavigateUp()
```
Это делается через переопределение метода onBackPressed()

severianremi commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -17,11 +17,18 @@
 
-    <android.support.v7.widget.RecyclerView
-        android:id="@+id/city_recyclerView"
+    <android.support.design.widget.TabLayout
```
Я просила сделать табы снизу, а не сверху =)

severianremi commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,8 +1,11 @@
+    <string name="share">Share</string>
+    <string name="city_page_title">Сities</string>
+    <string name="park_page_title">Parks</string>
+    <string name="Unknown">Unknown</string>
```
неправильное название

Valentindp committed:
1. Added context to fragment adapter
2. onBackPressed() for activity
3. Portrait orientation for CityActivity.kt and ParkActivity.kt

Valentindp committed:
Rename variable

Valentindp committed:
Rename variable

Commit:
Reformat code

Commit:
Parcelable for city and park class

Valentindp committed:
Add BottomBar

Commit:
BottomBar

Commit:
BottomBar fix

Commit:
add drawable

Commit:
implementation bottombar
